src/com/palmergames/bukkit/towny/event/town/TownMergeEvent.java

```diff
@@ -0,0 +1,37 @@
+package com.palmergames.bukkit.towny.event.town;
+
+import com.palmergames.bukkit.towny.object.Town;
+
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class TownMergeEvent extends Event {
+    private static final HandlerList handlers = new HandlerList();
+    
+    private final Town town;
```

Comment:
This event is thrown after the Town is removed so the finalTown will never exist at this point. It would be useful to give the name and UUID of the removed Town so things can be cleaned up in the plugin using this event.